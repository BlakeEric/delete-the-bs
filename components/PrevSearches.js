const PrevSearches = (props) => {

  if (props.data.length < 2) {
    return ''
  }

  return (
    <div className="prevSearches">
      <h4 className="prevSearches-header">Previous Searches</h4>
      <ul>
        {props.data.map((item, index) => {
          return index > 0 &&
          <li key={`prevSearch-${index}`}>
            <a href="#" onClick={(e) => {
              e.preventDefault
              props.handleResubmit(item)
            }}>{item}</a>
          </li>
        })}
      </ul>
    </div>
  )

}

export default PrevSearches

PrevSearches.defaultProps = {
  data: []
}
