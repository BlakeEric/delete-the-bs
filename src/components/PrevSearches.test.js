/* eslint-env jest */
import ReactDOM from 'react-dom'
import PrevSearches from './PrevSearches'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('PrevSearches component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PrevSearches />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders nothing when data is empty', () => {
    const wrapper = shallow(<PrevSearches />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders nothing when data array length is 1', () => {
    const wrapper = shallow(<PrevSearches data={["https://google.com"]}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders ui correctly when more than one item exits in data prop', () => {
    const wrapper = shallow(<PrevSearches data={["https://google.com", "https://wikipedia.com", "https://medium.com"]}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

});
