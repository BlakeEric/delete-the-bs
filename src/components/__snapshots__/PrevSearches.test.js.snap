// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevSearches component renders nothing when data array length is 1 1`] = `""`;

exports[`PrevSearches component renders nothing when data is empty 1`] = `""`;

exports[`PrevSearches component renders ui correctly when more than one item exits in data prop 1`] = `
<aside
  className="prevSearches"
>
  <h4
    className="prevSearches-header"
  >
    Previous Searches
  </h4>
  <ul>
    <li
      key="prevSearch-1"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        wikipedia.com
      </a>
    </li>
    <li
      key="prevSearch-2"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        medium.com
      </a>
    </li>
  </ul>
</aside>
`;
